src/components/icon-as-button/IconAsButton.jsx

```diff
@@ -1,22 +1,52 @@
+// @flow
 import React from 'react';
-import PropTypes from 'prop-types';
+import type {Node, Element} from 'react';
 import classNames from 'classnames';
-import Icon, {TYPE, ICON_COLOR} from '../icons/Icon';
+import * as IconModule from '../icons/Icon';
 import IconAsButtonContent from './subcomponents/IconAsButtonContent';
 
+const {ICON_COLOR, default: Icon} = IconModule;
+
+type IconSizeType =
+  | 'normal'
+  | 'small'
+  | 'xsmall'
+  | 'xxsmall';
+
 export const SIZE = {
-  NORMAL: {className: 'normal', iconSize: 26},
-  SMALL: {className: 'small', iconSize: 18},
-  XSMALL: {className: 'xsmall', iconSize: 14},
-  XXSMALL: {className: 'xxsmall', iconSize: 10}
+  NORMAL: 'normal',
+  SMALL: 'small',
+  XSMALL: 'xsmall',
+  XXSMALL: 'xxsmall'
+};
+
+const ICON_SIZE = {
+  [SIZE.NORMAL]: 26,
+  [SIZE.SMALL]: 18,
+  [SIZE.XSMALL]: 14,
+  [SIZE.XXSMALL]: 10
+};
+
+type IconAsButtonPropsType = {
+  size?: IconSizeType,
+  color?: IconModule.IconColorType,
+  type?: ?IconModule.IconTypeType,
+  children?: ?Element<any>,
```

Comment:
hmm why not `React.Node`?

Reply:
We should pass exactly one element, not an array, and `Element` is just like that, while `Node` can be an array or literal. We had such test before, you can check it :)